src/polynomial/grouped_poly.rs

```diff
@@ -43,6 +45,35 @@ impl<F: PrimeField> From<HashMap<usize, Expression<F>>> for GroupedPoly<F> {
 }
 
 impl<F: PrimeField> GroupedPoly<F> {
+    fn new(expr: Expression<F>, num_of_poly: usize, num_of_challenge: usize) -> Self {
+        expr.evaluate(
+            &|constant| GroupedPoly {
+                terms: vec![Some(Expression::Constant(constant))],
+            },
+            &|poly| {
+                let y = Expression::Polynomial(Query {
+                    index: poly.index + num_of_poly,
```

Comment:
Does the `y` here correspond to the fold transform: a -> a1+y*a2?

Reply:
Yes, I just take this name from current `fold_transform`